src/Plugin/AccessControlHierarchy/Taxonomy.php

```diff
@@ -328,7 +329,11 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
             $handler_settings = $field_definitions[$field_name]->getSetting('handler_settings');
             // Must refer to a proper target. Target bundles referring to
             // themselves would create an infinite loop. Deny.
-            if (!isset($handler_settings['target_bundles']) || in_array($bundle, $handler_settings['target_bundles'], TRUE)) {
+            if ($entity_type_id == 'taxonomy_term' && in_array($bundle, $this->configuration['vocabularies'])) {
```

Comment:
we should always use the third argument to in_array when comparing strings https://3v4l.org/vnfLY